tracking-service/src/main/java/com/tracking_delivery_system/tracking_service/service/DeliveryProgressSimulator.java

```diff
@@ -0,0 +1,54 @@
+package com.tracking_delivery_system.tracking_service.service;
+
+import com.tracking_delivery_system.tracking_service.model.LocationUpdate;
+import com.tracking_delivery_system.tracking_service.repository.LocationUpdateRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.*;
+
+@Service
+@RequiredArgsConstructor
+public class DeliveryProgressSimulator {
+    private final KafkaProducer producer;
+    private final LocationUpdateRepository repository;
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private final ConcurrentMap<UUID, ScheduledFuture<?>> tasksMap = new ConcurrentHashMap<>();
+    private Random random = new Random();
+
+    public void simulateDeliverProgress(LocationUpdate locationUpdate) {
+        final long[] distance = {locationUpdate.getDistance()};
```

Comment:
why you create array from string it in all subsequent steps you are always using only first element of this array (distance[0])?

Reply:
To allow the variable distance to be modified within the lambda expression.

Reply:
> why you create array from string it in all subsequent steps you are always using only first element of this array (distance[0])?

To allow the variable distance to be modified within the lambda expression.

Reply:
Interesting solution... But i think you could avoid this redundant complexity just having one variable inside of lambda "if" condition, besides you don't modify distance for lambda's "else" condition. So it's up to you, i don't have strict position against it